remote-api/src/main/java/org/alfresco/repo/web/scripts/content/ContentStreamer.java

```diff
@@ -496,6 +498,20 @@ public void setAttachment(WebScriptRequest req, WebScriptResponse res, boolean a
             res.setHeader("Content-Disposition", headerValue);
         }
     }
+
+    private String encodeFilename(String attachFileName)
+    {
+        try
+        {
+            return Rfc5987Util.encode(attachFileName);
+        } catch (UnsupportedEncodingException e)
+        {
+            if (logger.isInfoEnabled())
+                logger.info(e.getMessage() + " Changing encoder.");

```

Comment:
Do you think it's worth including the names of the previous and replaced encoders in this log message?

Personally, I prefer even a single `if` statement to be in its own block to make it easier to avoid mistakes later, but I think it's a preference issue, and I'm not going to argue with Robert Martin :)

Reply:
Added info about replaced encoders.
I will stay with current `if` for logger, to be compliant with rest of the class.